### PR TITLE
Add saturatef function

### PR DESCRIPTION
**Pull request by guansong**

### CODE REVIEW

File: cuda2gcn/src/math.cl [hunk 1]
```diff
@@ -279,12 +279,33 @@ ATTR double __nv_modf(double x, __private double *ptr) { return __ocml_modf_f64(
...
 //-------- T __nv_remquo
 ATTR double __nv_remquo(double x, double y, __private int *ptr) { return __ocml_remquo_f64(x, y, ptr); }
 
 //-------- T __nv_remquof
 ATTR float __nv_remquof(float x, float y, __private int *ptr) { return __ocml_remquo_f32(x, y, ptr); }
 
+//-------- T __nv_saturatef
+ATTR float __nv_saturatef(float x)
+{
+    if (isnan(x)) return 0;
```
Comment (b-sumner) [inline cuda2gcn/src/math.cl]: The body should be "return __ocml_min_f32(__ocml_max_f32(x, 0.0f), 1.0f);"

**Comment by guansong** [comment]
thanks. updated to proper const format.

File: cuda2gcn/src/math.cl [hunk 2]
```diff
@@ -279,12 +279,39 @@ ATTR double __nv_modf(double x, __private double *ptr) { return __ocml_modf_f64(
...
 //-------- T __nv_remquo
 ATTR double __nv_remquo(double x, double y, __private int *ptr) { return __ocml_remquo_f64(x, y, ptr); }
 
 //-------- T __nv_remquof
 ATTR float __nv_remquof(float x, float y, __private int *ptr) { return __ocml_remquo_f32(x, y, ptr); }
 
+//-------- T __nv_saturatef
+ATTR float __nv_saturatef(float x)
+{
+    if (isnan(x)) return 0.0f;
```
Comment (b-sumner) [inline cuda2gcn/src/math.cl]: This function is a simple clamp.  The entire body can be simply written as return __ocml_min_f32(__ocml_max_f32(x, 0.0f), 1.0f);

**Comment by guansong** [comment]
I see. Will update it.

**Comment by b-sumner** [comment]
I already merged many of these.  Can you generate a new pull request against the latest?  Also, if you can wait a day or two, we should have rhypot, rlen3, and rlen4 checked in to OCML.

**Pull request merged**